packages/medusa/src/models/order-edit.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  BeforeInsert,
+  Column,
+  CreateDateColumn,
+  JoinColumn,
+  ManyToOne,
+  OneToMany,
+} from "typeorm"
+
+import OrderEditingFeatureFlag from "../loaders/feature-flags/order-editing"
+import { FeatureFlagEntity } from "../utils/feature-flag-decorators"
+import { resolveDbType } from "../utils/db-aware-column"
+import { OrderItemChange } from "./order-item-change"
+import { SoftDeletableEntity } from "../interfaces"
+import { generateEntityId } from "../utils"
+import { LineItem } from "./line-item"
+import { Order } from "./order"
+
+@FeatureFlagEntity(OrderEditingFeatureFlag.key)
+export class OrderEdit extends SoftDeletableEntity {
+  @Column({ nullable: true })
+  order_id: string
+
+  @ManyToOne(() => Order, (o) => o.edits)
+  @JoinColumn({ name: "order_id" })
+  order: Order
+
+  @OneToMany(() => OrderItemChange, (oic) => oic.order_edit, {
+    cascade: true,
+  })
+  changes: OrderItemChange[]
+
+  @Column({ nullable: true })
+  internal_note?: string
+
+  @Column({ nullable: true })
+  created_by: string // customer or user ID
+
+  @CreateDateColumn({ type: resolveDbType("timestamptz") })
+  created_at: Date
+
+  @Column({ nullable: true })
+  requested_by?: string // customer or user ID
+
+  @CreateDateColumn({ type: resolveDbType("timestamptz"), nullable: true })
+  requested_at?: Date
+
+  @Column({ nullable: true })
+  confirmed_by?: string // customer or user ID
+
+  @CreateDateColumn({ type: resolveDbType("timestamptz"), nullable: true })
```

Comment:
these should not be `@CreateDateColumn` since that decorator type is a date column, but is set to `Date.now()` on insertion time: [CreateDateColumn](https://orkhan.gitbook.io/typeorm/docs/decorator-reference#createdatecolumn)

Instead use a regular column like in the batch job model: 
`@Column({ type: resolveDbType("timestamptz"), nullable: true })`

The migrations should also be updated to reflect this change as they now set the default value of the column to now: 
`"confirmed_at" TIMESTAMP WITH TIME ZONE DEFAULT now(),`